_posts/2017-05-15-rust-at-two-years.md

```diff
@@ -0,0 +1,356 @@
+
+[incrcomp]: https://internals.rust-lang.org/t/incremental-compilation-beta/4721
+[roadmap-4]: https://github.com/rust-lang/rust-roadmap/issues/4
+
+### [Rust should provide a basic, but solid IDE experience](https://github.com/rust-lang/rust-roadmap/issues/2)
+
+As part of our IDE initiative, we created the [Rust Language Server] project.
+Its goal is to create a single tool that makes it easy for any editor or IDE to
+have the full power of the Rust compiler for error checking, code navigation,
+and refactoring by using the standard [language server
+protocol](http://langserver.org/) created by Microsoft and Eclipse.
+
+While still early in its life, today the RLS is [available from rustup] for
+nightly users. It provides type information on hover, error messages as you
+type, and different kinds of code navigation. It even provides refactoring and
+formatting as unstable features! It works with projects as large as Cargo.
+We're excited to watch the RLS continue to grow and hope to see it make its way
+to stable Rust later this year.
+
+[Rust Language Server]: https://github.com/rust-lang-nursery/rls
+[available from rustup]: https://github.com/rust-lang-nursery/rls#setup
+
+*Thanks to Jonathan Turner for this RLS summary!*
+
+### [Rust should have 1.0-level crates for essential tasks](https://github.com/rust-lang/rust-roadmap/issues/11), and [Rust should provide easy access to high quality crates](https://github.com/rust-lang/rust-roadmap/issues/9)
+
+The [recent post on the Libz Blitz] details the Library Team's initiative to
+increase the quality of crates for common tasks; that post is excellent so I
+won't repeat it here. I will note that many of the issues that the Libs Team
+is going to create will be great starter issues. For the blitz to be the best
+it can be, the Libs Team is going to need help from the community-- that means
+YOU! :) They're willing to mentor people interested in contributing.
+
+In order to make awesome crates easier to find for particular purposes,
+crates.io now has [categories] for crate authors to better indicate the use
+case of their crate. Crates can also now have CI badges, and
+[more improvements to crates.io's interface] are coming that will help you
+choose the crates that fit your needs.
+
+[recent post on the Libz Blitz]: https://blog.rust-lang.org/2017/05/05/libz-blitz.html
+[categories]: https://crates.io/categories
+[more improvements to crates.io's interface]: https://github.com/rust-lang/rust/issues/41616
+
+### [Rust should be well-equipped for writing robust, high-scale servers](https://github.com/rust-lang/rust-roadmap/issues/10)
+
+TODO: [Futures] and [tokio] enable zero-cost asynchronous programming
+
+[Futures]: https://crates.io/crates/futures
+[tokio]: https://tokio.rs/
+
+### [Rust should integrate easily into large build systems](https://github.com/rust-lang/rust-roadmap/issues/12)
+
+This initiative is mostly in the ideas stage; there's lots of great discussion
```

Comment:
Could expand this slightly to say: we've done a lot of work with stakeholders to understand the challenges in build system integration today, and think we have a good overall vision for how to solve them, but there are a lot of details to be worked out.